ee/control/control.go

```diff
@@ -0,0 +1,148 @@
+package control
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+)
+
+// ControlService is the main object that manages the control service. It is responsible for fetching
+// and caching control data, and updating consumers and subscribers.
+type ControlService struct {
+	logger          log.Logger
+	cancel          context.CancelFunc
+	requestInterval time.Duration
+	consumers       map[string]consumer
+	subscribers     map[string][]subscriber
+	fetcher         dataProvider
+	lastFetched     map[string]string
+}
+
+// consumer is an interface for something that consumes control server data updates. The
+// control server supports at most one consumer per subsystem.
+type consumer interface {
+	Update(io.Reader)
+}
+
+// subscriber is an interface for something that wants to be notified when a subsystem has been updated.
+// Subscribers do not receive data -- they are expected to read the data from where consumers write it.
+type subscriber interface {
+	Ping()
+}
+
+// dataProvider is an interface for something that can retrieve control data. Authentication, HTTP,
+// file system access, etc. lives below this abstraction layer.
+type dataProvider interface {
+	Get(hash string) (data io.Reader, err error)
+}
+
+func New(logger log.Logger, data dataProvider, opts ...Option) *ControlService {
+	cs := &ControlService{
+		logger:          logger,
+		requestInterval: 60 * time.Second,
+		consumers:       make(map[string]consumer),
+		subscribers:     make(map[string][]subscriber),
+		fetcher:         data,
```

Comment:
```suggestion
		fetcher:         fetcher,
```

(Or leave till the next iteration